staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -830,9 +835,21 @@ func (s *Controller) lockedUpdateLoadBalancerHosts(service *v1.Service, hosts []
 	return err
 }
 
-func wantsLoadBalancer(service *v1.Service) bool {
-	// if LoadBalancerClass is set, the user does not want the default cloud-provider Load Balancer
-	return service.Spec.Type == v1.ServiceTypeLoadBalancer && service.Spec.LoadBalancerClass == nil
+func wantsLoadBalancer(service *v1.Service, allowedLoadbalancerClass []*string) bool {
+	if service.Spec.Type == v1.ServiceTypeLoadBalancer {
+		if service.Spec.LoadBalancerClass == nil {
+			// Default provider should accept svc, and other provider should ignore
+			return len(allowedLoadbalancerClass) == 0
```

Comment:
Shouldn't this just return `true`?

If service.Spec.LoadBalancerClass is nil, which means we want to use the cloud provider's load balancer implementation, no matter there is `allowedLoadbalancerClass` or not, right?

Reply:
Imagine there are a few cloud providers available. the default provider will handle all svc whose type is ServiceTypeLoadBalancer and LoadBalancerClass  is nil. allowedLoadbalancerClass  is empty if this informer is created by default provider
other providers only handle svc whose LoadBalancerClass  is not nil. allowedLoadbalancerClass  is not empty, the provider here will only handle request whose LoadBalancerClass  falls in allowedLoadbalancerClass  

IF LoadBalancerClass   is nil, This svc should be ignored by other provider and handled by the default provider